File: answers/leetcode/Climbing Stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """

        dp = [0 for i in range(n + 1)]
        dp[0] = 1
        for i in range(n):
            if i + 1 < len(dp):
                dp[i + 1] += dp[i]
            if i + 2 < len(dp):
                dp[i + 2] += dp[i]
        return dp[-1]

File: answers/other/answers/lintcode/Climbing Stairs.cpp
class Solution {
public:
    /**
     * @param n: An integer
     * @return: An integer
     */
    int climbStairs(int n) {
        // write your code here
		vector<int> dp(n + 1, 0);
		dp[0] = 1;
		for (size_t i = 0; i <= n; ++ i) {
			if (dp[i]) {
				if (i + 1 <= n) {
					dp[i + 1] += dp[i];
				}
				if (i + 2 <= n) {
					dp[i + 2] += dp[i];
				}
			}
		}
		return dp[n];
    }
};

